test/suite/test_timestamp14.py

```diff
@@ -0,0 +1,244 @@
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+# test_timestamp14.py
+#   Global timestamps: oldest reader, all committed, pinned
+#
+
+import random
+from suite_subprocess import suite_subprocess
+import wiredtiger, wttest
+from wtscenario import make_scenarios
+
+def timestamp_str(t):
+    return '%x' % t
+
+class test_timestamp14(wttest.WiredTigerTestCase, suite_subprocess):
+    tablename = 'test_timestamp14'
+    uri = 'table:' + tablename
+
+    def test_all_committed(self):
+        all_committed_uri = self.uri + '_all_committed'
+        session1 = self.setUpSessionOpen(self.conn)
+        session2 = self.setUpSessionOpen(self.conn)
+        session1.create(all_committed_uri, 'key_format=i,value_format=i')
+        session2.create(all_committed_uri, 'key_format=i,value_format=i')
+
+        # Scenario 0: No commit timestamp has ever been specified therefore
+        # There is no all_committed timestamp and we will get an error
+        # Querying for it.
+        session1.begin_transaction()
+        cur1 = session1.open_cursor(all_committed_uri)
+        cur1[1]=1
+        session1.commit_transaction()
+        self.assertRaisesException(wiredtiger.WiredTigerError,
+            lambda: self.conn.query_timestamp('get=all_committed'))
+
+        # Scenario 1: A single transaction with a commit timestamp, will
+        # result in the all_committed timestamp being set.
+        session1.begin_transaction()
+        cur1[1]=1
+        session1.commit_transaction('commit_timestamp=1')
+        self.assertTimestampsEqual(
+            self.conn.query_timestamp('get=all_committed'), "1")
+
+        # Scenario 2: A transaction begins and specifies that it intends
+        # to commit at timestamp 2, a second transaction begins and commits
+        # at timestamp 3.
+        session1.begin_transaction()
+        session1.timestamp_transaction('commit_timestamp=2')
+
+        session2.begin_transaction()
+        cur2 = session2.open_cursor(all_committed_uri)
+        cur2[2] = 2
+        session2.commit_transaction('commit_timestamp=3')
+
+        # As the original transaction is still running the all_committed
+        # timestamp is being held at 1.
+        self.assertTimestampsEqual(
+            self.conn.query_timestamp('get=all_committed'), "1")
+        cur1[1] = 2
+        session1.commit_transaction()
+
+        # Now that the original transaction has finished the all_committed
+        # timestamp has moved to 3, skipping 2 as there is a commit with
+        # a greater timestamp already existing.
+        self.assertTimestampsEqual(
+            self.conn.query_timestamp('get=all_committed'), "3")
+
+        # Senario 3: Commit with an all committed of 5 and then begin a
+        # transaction intending to commit at 4, the all_committed timestamp
+        # should move back.
+        session1.begin_transaction()
+        cur1[1] = 3
+        session1.commit_transaction('commit_timestamp=5')
```

Comment:
commit timestamp is inconsistent with the comment, code is inconsistent with scenario specified above.